pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -203,16 +207,38 @@ public static void main(String[] args) {
     }
 
     public static void start(String[] args, SendGridSupplier sendGridSupplier) {
+        boolean housekeepingLocalSetup = false;
+        boolean usePubSubEmulator = false;
+        try {
+            Options options = new Options();
+            options.addOption(null, "local-housekeeping", false, "starts housekeeping as local service");
+            options.addOption(null, "emulator", false, "starts housekeeping as local emulator client");
+            CommandLineParser parser = new DefaultParser();
+            CommandLine cmd = parser.parse(options, args);
+            housekeepingLocalSetup = cmd.hasOption("local-housekeeping");
+            usePubSubEmulator = !housekeepingLocalSetup && cmd.hasOption("emulator");
+        } catch (Exception e) {
+            log.info("command line parameters parse error");
+        }
+
         LogbackConfigurationPrinter.printLoggingConfiguration();
         Config cfg = ConfigManager.getInstance().getConfig();
         boolean doLiquibase = cfg.getBoolean(ConfigFile.DO_LIQUIBASE);
         int maxConnections = cfg.getInt(ConfigFile.HOUSEKEEPING_NUM_POOLED_CONNECTIONS);
         String pubSubProject = cfg.getString(ConfigFile.GOOGLE_PROJECT_ID);
 
-        boolean usePubSubEmulator = cfg.getBoolean(ConfigFile.USE_PUBSUB_EMULATOR);
-        String housekeepingDbUrl = cfg.getString(TransactionWrapper.DB.HOUSEKEEPING.getDbUrlConfigKey());
+        if (!usePubSubEmulator) {
+            usePubSubEmulator = !housekeepingLocalSetup && cfg.getBoolean(ConfigFile.USE_PUBSUB_EMULATOR);
+        }
         String apisDbUrl = cfg.getString(TransactionWrapper.DB.APIS.getDbUrlConfigKey());
 
+        final String housekeepingDbUrl = (usePubSubEmulator || housekeepingLocalSetup)
+                ? apisDbUrl
+                : cfg.getString(TransactionWrapper.DB.HOUSEKEEPING.getDbUrlConfigKey());
+        final String proxy = (usePubSubEmulator || housekeepingLocalSetup)
+                ? null
+                : ConfigUtil.getStrIfPresent(cfg, ConfigFile.Sendgrid.PROXY);
+
```

Comment:
Why is this preferable to removing the proxy configuration from the configuration file?

Reply:
> Why is this preferable to removing the proxy configuration from the configuration file?

we can not access sendgrid via proxy from local environment

Reply:
I understand, but that's not what I was asking.

The proxy will only be used if it is the `sendgrid.proxy` key set to a non-blank value- why does this need a flag? If a flag is desirable in this case, let's override the value in the configuration instead.

In addition, the `dev` proxy _is_ accessible from a local environment when connected to the Broad VPN.